src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -772,6 +772,151 @@ def hcells_list(self, value: List[str]) -> None:
+        try:
+            return self.attr_getter("_top_module", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the top RTL module yet")
+
+    @top_module.setter
+    def top_module(self, value: str) -> None:
+        """Set the top RTL module."""
+        if not (isinstance(value, str)):
+            raise TypeError("top_module must be a str")
+        self.attr_setter("_top_module", value)
+ 
+    @property
+    def options(self) -> List[str]:
+        """
+        Get the additional simulation options.
+
+        :return: The list of additional simulation options.
+        """
+        try:
+            return self.attr_getter("_options", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the additional simulation options")
+
+    @options.setter
+    def options(self, value: List[str]) -> None:
+        """Set the additional simulation options."""
+        if not (isinstance(value, List[str])):
+            raise TypeError("options must be a List")
+        self.attr_setter("_options", value)
+
+    @property
+    def debug(self) -> bool:
+        """
+        Get the debug flag.
+
+        :return: The debug flag boolean.
+        """
+        try:
+            return self.attr_getter("_debug", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the simulation debug flag")
+
+    @debug.setter
+    def debug(self, value: bool) -> None:
+        """Set the simulation input files."""
+        if not (isinstance(value, bool)):
+            raise TypeError("debug must be a bool")
+        self.attr_setter("_debug", value)
+
+    @property
+    def defines(self) -> List[str]:
+        """
+        Get the simulation defines.
+
+        :return: The list of simulation defines.
+        """
+        try:
+            return self.attr_getter("_defines", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the simulation defines")
+
+    @defines.setter
+    def defines(self, value: List[str]) -> None:
+        """Set the simulation defines."""
+        if not (isinstance(value, List[str])):
+            raise TypeError("defines must be a List")
+        self.attr_setter("_defines", value)
+
+    @property
+    def compiler_opts(self) -> List[str]:
+        """
+        Get the simulation compiler options.
+
+        :return: The list of simulation compiler options.
+        """
+        try:
+            return self.attr_getter("_compiler_opts", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the simulation comiler options")
+
+    @compiler_opts.setter
+    def compiler_opts(self, value: List[str]) -> None:
+        """Set the simulation compiler options."""
+        if not (isinstance(value, List[str])):
+            raise TypeError("compiler_opts must be a List")
+        self.attr_setter("_compiler_opts", value)
+
+    @property
+    def timescale(self) -> str:
```

Comment:
Can we use a `TimeValue` instead of a raw `str` here?

Reply:
I agree, although it should be a tuple of `TimeValue`s (unit and resolution). e.g. `(TimeScale("1 ns"), TimeScale("1 ps"))` for `` `timescale 1ns / 1ps``

A `NamedTuple` would also work.